### DISCUSSION TITLE
Disable dark-mode (class method) for specific elements

### DISCUSSION THREAD

**Discussion by AhmedFarhan252** [tailwindlabs/tailwindcss#8314] [Help] [answered]
Is there any way to disable dark mode's `dark:` formatting for specific elements? 

Let's say I have a DOM where the root is passed the `dark` class. Now, I don't want the dark mode styling to be used for a specific child element as well as the children of that child. So preferably, I add a class like `light` to that child and that disables all `dark:` stylings in that child and its children.

**Comment by carlos-dubon** [comment]
Just remove the `dark` attributes from those elements. This is the only way to achieve what you want.

**Reply by AhmedFarhan252** [reply]
The `dark` attribute has been added to the root of the document. Is it possible to ignore/remove it for specific elements, since I did not see it when I inspected the attributes of the children nodes?

**Comment by rohmann** [comment, marked as answer]
Hi @AhmedFarhan252 ,

I think this is something better managed in how the HTML is setup. Trying to "undo" styles in CSS at scale gets messy, and it's like trying to paddle upstream, against it's cascading nature. If you set `dark` on the root, anything that unsets that needs to add a bunch of extra styles to fight against it.

I know many designs rely on alternating light or dark sections, or reusing design elements. I would just try to account for that in the HTML like:

```html
<body>
  <header></header>
  <section class="dark"></section>
  <section class="dark"></section>
  <section></section>
  <section class="dark"></section>
  <section class="dark"></section>
  <section></section>
</body>
```

If you're using a framework or any kind of state manager, it should still be possible to logically manage a light/dark toggle in a single place globally (like React Context) but applying your dark mode more granularly to portions of your site will yield cleaner CSS output.

**Reply by AhmedFarhan252** [reply]
Thanks! 
I'll go with this considering the nature of css
